resources/install.rb

```diff
@@ -20,15 +20,54 @@
 include IISCookbook::Helper
 
 property :source, String
-property :additional_components, Array, default: []
+property :additional_components, [Array, String],
+ coerce: proc { |c| Array(c) },
+ default: []
+property :install_method, [String, Symbol],
+  required: false,
+  coerce: proc { |i| i.to_sym },
+  equal_to: [:windows_feature_dism, :windows_feature_powershell, :windows_feature_servermanagercmd],
+  default: :windows_feature_dism
+property :start_iis, [true, false], default: false
```

Comment:
Why are we controlling the service in the install resource now?

Reply:
Every recipe in the cookbook was doing the same things really.

Include the default recipe, add another windows feature and then sometimes one other resource.

The default recipe would install iis and then start it.

Moving the starting into the resource (bare in mind it's optional and doesn't by default to keep past behaviour the same for users of the resource) meant that 

1. Every recipe could now call the install resource and pass its own additional features along with, removing the need to use windows_feature in almost every recipe.
2. Now there was only one place windows_feature is being used. The install resource, this allows us to keep the 'translation of feature names' just contained to that resource.
3. All the recipes are dead simple now and if at some point in the future sous-chefs want to remove them it will be much easier for people to transition and keep the same behaviour

Reply:
It's been a while since i've had to use IIS but i'm pretty sure the w3svc service is enabled by default when IIS is installed? So I don't think this is necessary. Also, it's adding unexpected functionality to the resource, I don't expect an install resource to do anything other than install.

If people need to manage the IIS service then it's not a big deal to do that in the implementing wrapper as there's no abstraction required outside of the core Chef resources.

Reply:
It doesn't start it by default you have to set the property.
The reason I added it was it allows the rest of the recipes to be much simpler, since they all called the default recipe anyway and the default recipe started the service and I wanted to ensure I wasn't changed anything for existing people who don't want to have to update how they use this cookbook. It means that all recipes rely on the install resource to install their features and we can keep that feature name translation in one place rather than having to spread it around the cookbook.

For outside users the install resource really hasn't changed if they don't want. By default it won't try and start the service (which according yourself is probably done anyway), personally I think this gives us the benefit of new functionality with out breaking any existing features/APIs, whilst also laying the ground work for the future desire of just exposing resources rather than these recipes. The recipes are now very basic so it's easy for people to 're-implement' them in any future breaking change